mpl/_mpl.py

```diff
@@ -0,0 +1,44 @@
+import torch
+
+def compute_weights(losses, indices, weights, ratio, p):
+    size = losses.size(0)
+
+    # find first nonzero element
+    pos = 0
+    while losses[pos]< 1e-5:
+        pos += 1
+    n = size - pos
+    m = int(ratio * n)
+    if n <= 0 or m <= 0:
+        raise ValueError
+    q = p / (p - 1.0)
+    c = m - n + 1
+    a = [0.0 , 0.0]
+    i = pos
+    nu = 0.0
+    while i < n and nu < 1e-5:
```

Comment:
Oh, the main reason for implement compute_weight in C++ is that python's loops are too slow and this loop will be a too slow for large batch size. Can you provide performance report which show that python code isn't slower than c++ implementation?

Reply:
Good point. However i haven't compared yet. Thanks for your patience again.

Reply:
I cann't run build.py.  So i  try the python version , but it doesn't work. Have you succeed?

Reply:
@01lin currently this implementation is not compatible with pytorch 0.5+

Reply:
@BelBES  With pytorch 0.4.0  and python3,  I also cann't run build.py. And the error is  " cffi.error.VerificationError: CompileError: command 'gcc' failed with exit status 1".  Can you help me?